code/modules/robotics/bot/guardbot.dm

```diff
@@ -971,31 +1468,42 @@
-				if(src.budgun)
-					shotcount = 1	// TODO: Make rapidfire exist, then work.
-					while(shotcount > 0 && target)
+				if (istype(src.budgun, /obj/item/bang_gun))
+					src.budgun.pixelaction(target, null, src, null) // dang it
+					GunSux()
+				else if(istype(src.budgun, /obj/item/gun/russianrevolver))
+					BarGun()
+				else if(src.budgun)
+					var/burst = shotcount	// TODO: Make rapidfire exist, then work.
+					while(burst > 0 && target)
 						budgun.shoot(get_turf(target), get_turf(src), src)
-						shotcount--
-
+						burst--
+						if (burst)
+							sleep(5)	// please dont fuck anything up
 					if (istype(src.budgun, /obj/item/gun/kinetic))
 						var/obj/item/gun/kinetic/shootgun = src.budgun
-						if (ammofab)	// Can we build our own ammo?
-							if (shootgun.ammo && shootgun.ammo.max_amount && (shootgun.ammo.amount_left <= 1))
-								if (cell.charge && (cell.charge >= GUARDBOT_LOWPOWER_ALERT_LEVEL) && ((cell.charge - ((shootgun.ammo.max_amount - shootgun.ammo.amount_left) * (shootgun.ammo.ammo_type.power * shootgun.ammo.ammo_type.ks_ratio * 2))) > (0.2 *GUARDBOT_LOWPOWER_ALERT_LEVEL)))	// *scream
-									cell.charge -= ((shootgun.ammo.max_amount - shootgun.ammo.amount_left) * (shootgun.ammo.ammo_type.power * shootgun.ammo.ammo_type.ks_ratio * 2))
-									shootgun.ammo.amount_left = shootgun.ammo.max_amount
+						if (ammofab || istype(src.budgun, /obj/item/gun/kinetic/meowitzer/inert))	// Can we build our own ammo?
+						//	if (shootgun?.ammo?.max_amount && (shootgun.ammo.amount_left <= 1 || ((shootgun.ammo.amount_left - shootgun.current_projectile.cost) <= 0))) // was making this useless with guns that you'd actually want to use it, like the AK. Wouldnt even reload!
+							if (cell.charge && (cell.charge >= GUARDBOT_LOWPOWER_ALERT_LEVEL) && ((cell.charge - ((shootgun.ammo.max_amount - shootgun.ammo.amount_left) * (shootgun.ammo.ammo_type.power * shootgun.ammo.ammo_type.ks_ratio * 0.75))) > (GUARDBOT_LOWPOWER_ALERT_LEVEL)))	// *scream
+								cell.charge -= ((shootgun.ammo.max_amount - shootgun.ammo.amount_left) * (shootgun.ammo.ammo_type.power * shootgun.ammo.ammo_type.ks_ratio * 0.75))
+								shootgun.ammo.amount_left = shootgun.ammo.max_amount
 					else if (istype(src.budgun, /obj/item/gun/energy))
 						var/obj/item/gun/energy/pewgun = src.budgun
 						if (pewgun.cell && (pewgun.cell.charge < pewgun.cell.max_charge))
 							if (cell.charge && (cell.charge >= GUARDBOT_LOWPOWER_ALERT_LEVEL))
 								cell.charge -= (pewgun.cell.max_charge - pewgun.cell.charge)
 								pewgun.cell.charge = pewgun.cell.max_charge
+					else if (istype(src.budgun, /obj/item/gun/bling_blaster))	// It prints MONEY
+						var/obj/item/gun/bling_blaster/funds = src.budgun
+						if (funds.cash_amt && (funds.cash_amt < funds.cash_max))
+							if (cell.charge && (cell.charge >= GUARDBOT_LOWPOWER_ALERT_LEVEL))
+								cell.charge -= (funds.cash_max - funds.cash_amt)
+								funds.cash_amt = funds.cash_max
```

Comment:
😕

Reply:
Changed to require the ammofab to do that.

I also did some (maybe accurate?) math to see how much, on average, an infinite-ammo bling blaster on a Buddy would make:

BUDDYCHARGE = 1500 (usually around 1400ish, but lets say they're freshly charged and don't have a passive drain.)
CHARGE PER BLING SHOT = 100
= 15 shots before the buddy is fully drained (10 shots if not ammofabbed)
4 seconds per shot
=
1 minute to drain the Buddy,
1 minute to shoot 15 times.

BLING chance to shoot RARE = 10%
BLING chance to shoot RARE and GEMSTONE = 5%
BLING chance to shoot RARE and GEMSTONE is VERY VALUABLE = 0.5%
VALUE of VERY VALUABLE GEMSTONE = 700
EV from GEM = 52c / min
BLING chance to shoot RARE and GEMSTONE is VALUABLE = 1.5%
VALUE of VERY VALUABLE GEMSTONE = 500
EV from GEM = 68c / min
BLING chance to shoot RARE and GEMSTONE is SHIT = 3%
VALUE of VERY VALUABLE GEMSTONE = 200
EV from GEM = 90c / min
BLING chance to shoot RARE and GOLD = 5%
VALUE of GOLD NUGGET = 300
EV from GOLD = 225c / min !

BLING chance to shoot UNCOMMON = 18% (only rolls 1/5 if above 1/10 didn't pass.)
BLING chance to shoot UNCOMMON and HUNDRED BUX = 9%
VALUE of HUNDRED BUX = 100 :3
EV from HUNDRED BUX = 135c / min !
BLING chance to shoot UNCOMMON and COIN = 9%
VALUE of COIN = NOTHING! probably. lets say 1. vOv
EV from HUNDRED BUX = 1.35c / min

BLING chance to shoot COMMON = 72% (only rolls if above 1/5 and above 1/10 didn't pass.)
BLING chance to shoot COMMON and TENBUX = 21.6%
VALUE of TENBUX = 10 :3
EV from TENBUX = 16.25c / min
BLING chance to shoot COMMON and FIVE BUX = 21.6%
VALUE of FIVE BUX = 5 :3
EV from FIVE BUX = 8.125c / min !
BLING chance to shoot COMMON and BUCK = 21.6%
VALUE of BUCK = 5 :3
EV from BUCK = 1.625c / min !

Avg $$$/minute of BlingBuddy = 597.35 (600c)

all the while, 
the Buddy is pelting someone with cash (which apparently does brute damage),
you spent 4 TC on this,
you're picking up all this shit and selling the non-cash things,
you're babysitting this thing (unless you hooked a monkey up to an omnizine drip or something)
you made 750c from your paycheck.

If someone *really* wants to use this to farm cash (instead of just, like, roguing the AI into giving you the station budget, which as the RD you totally can even if you're not a badguy), I feel they should be allowed to.

Course you could give em a bigger battery, maybe something self-charging. Doesn't really change much other than that they dont need to recharge as much.

(also shows that the bling blaster, on average, eats more money than it gives)

Reply:
lol 👍